Fix trilogy builds

eileencodes opened the pull request:
https://github.com/rails/rails/pull/48007 and https://github.com/rails/rails/pull/47993 combined

eileencodes committed:
Fix database configuration and generation

I tried using the same config for mysql2 and trilogy but instead it
eneded up breaking the trilogy tests - they were only running in mysql2
mode. I wanted to do this so that the rake tasks for trilogy wouldn't
need to be duplicated but since that didn't work out quite right, I've
decide to duplicate the calls and add if exists / if not exists where
applicable. The configs should be the same but this will make sure that
if they do deviate, the dbs are always created/dropped.

eileencodes committed:
Raise an error if the connection name doesn't match the adapter

This will prevent us from accidentally running mysql2 tests against
trilogy adapter as we currently are doing.

eileencodes committed:
Bump trilogy to fix connection error translation

Until we do a new release we need to run against main on trilogy to get
the latest changes from https://github.com/github/trilogy/pull/69

eileencodes committed:
Fix config and copy over trilogy schema

We had a mysql2_specific_schema.rb file but not one for trilogy that was
causing tests to break. I also removed the prepared statements from
trilogy because it is not supported.

byroot committed:
TrilogyAdapterTest: use the proper config

byroot commented:
Uh?? Why is it still failing? I don't get it

casperisfine commented:
The paths are lies:

```
Error:
TrilogyAdapterTest#test_#reset_closes_connection_with_existing_connection:
Trilogy::SyscallError::ENOENT: No such file or directory - trilogy_connect - unable to connect to /tmp/mysql.sock
    /usr/local/bundle/bundler/gems/trilogy-6dd14cca18da/contrib/ruby/lib/trilogy.rb:174:in `_initialize'
    /usr/local/bundle/bundler/gems/trilogy-6dd14cca18da/contrib/ruby/lib/trilogy.rb:174:in `initialize'
    /rails/activerecord/test/cases/adapters/trilogy/trilogy_adapter_test.rb:169:in `new'
    /rails/activerecord/test/cases/adapters/trilogy/trilogy_adapter_test.rb:169:in `block in <class:TrilogyAdapterTest>'
 
bin/rails test /rails/activerecord/test/cases/adapters/trilogy/trilogy_adapter_test.rb:168
```

I don't get what's up.

byroot commented on activerecord/test/cases/adapters/trilogy/trilogy_adapter_test.rb:
```diff
@@ -9,11 +9,7 @@
-      username: "rails",
-      database: "activerecord_unittest",
-    }
+    @configuration = ARTest.config.fetch("connections").fetch("trilogy").fetch("arunit").symbolize_keys
```
I was missing the `symbolize_keys` 🤦

byroot commented:
Ok, so it now passed on 3.0 and 3.2, but failed for new reasons on 3.1 and 2.7.

casperisfine commented:
> failed for new reasons on 3.1 and 2.7.

I can repro locally by using the same seed. So it must be some kid of leak.

byroot committed:
Delete incompatible Trilogy tests

The trilogy test suite was imported directly, but many of
the test can't work as is in Rails test suite because they were
mutating DB state.

Most are redundant anyway, or should be implemented as shared test
that are executed for all adapters.

byroot commented:
Ok, so a lot of the tests imported from the gem weren't safe to run as part of the test suite because they mutate the database.

Most were redundant anyway so I just removed all the leaky tests.
